### PR TITLE
Fix wrong URL encoding in watcher HTTP client

### LINKED ISSUE #44970: Watcher webhook actions returning 302 response code in 7.2 and above.
**Elasticsearch version** (`bin/elasticsearch --version`): 7.2, 7.3

**Plugins installed**: []

**JVM version** (`java -version`): Bundled JDK

**OS version** (`uname -a` if on a Unix-like system): ESS and Darwin Kernel Version 18.6.0

**Description of the problem including expected versus actual behavior**: Watcher web hook actions do not return a 200 status code for POST requests.

**Steps to reproduce**: Execute a watch with a webhook action.

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Create watch with a webhook action.
 2. Force execute watch, or wait for it to trigger

As of `7.2`, webhook actions are are returning a `302` http status code for POST requests, as opposed to `200`. After some initial discussions with @jakelandis, there is nothing obvious in the codebase that would cause this.

The same exact webhook action in `7.1` and below works fine. All other http clients (tested with curl, wget and postman) all return the expected response. 

This has been tested (an initially spotted) in ESS, and also on my MBP.

### PR DESCRIPTION
**Pull request by spinscale**
The test assumption was calling the wrong method resulting in a URL
encoding before returning the data.

Closes #44970 (for real this time)

Reviewers note: This needs to be pushed to master, 7.x and 7.3

### CODE REVIEW

**Review by jakelandis — CHANGES_REQUESTED** [review]

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/http/HttpClient.java [hunk 1]
```diff
@@ -334,12 +334,11 @@ private HttpProxy getProxyFromSettings(Settings settings) {
                     String part = pathParts[i];
                     boolean isLast = i == pathParts.length - 1;
                     if (Strings.isEmpty(part) == false) {
-                        String appendPart = part;
+                        unescapedPathParts.add(URLDecoder.decode(part, StandardCharsets.UTF_8.name()));
                         boolean appendSlash = isPathEndsWithSlash && isLast;
                         if (appendSlash) {
-                            appendPart += "/";
+                            unescapedPathParts.add("");
```
[the remark this answers is not in the document]
Comment (spinscale) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/http/HttpClient.java]: yes. added a comment to clear things up

**Review by jakelandis — APPROVED** [review]
LGTM

**Comment by spinscale** [comment]
@elasticmachine update branch

**Comment by spinscale** [comment]
@elasticmachine run tests elasticsearch-ci/2

**Comment by spinscale** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**